ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/MiniProtocol/ObjectDiffusion/ObjectPool/PerasCert.hs

```diff
@@ -104,11 +113,30 @@ validatePerasCerts ::
   [PerasCert blk] ->
   m [ValidatedPerasCert blk]
 validatePerasCerts certs = do
-  let perasCfg = makePerasCfg Nothing
-  -- TODO replace the mocked-up Nothing with a real
-  -- 'BlockConfig' when all the plumbing is in place
+  let perasParams = mkPerasParams
+  -- TODO pass down 'BlockConfig' when all the plumbing is in place
   -- see https://github.com/tweag/cardano-peras/issues/73
   -- see https://github.com/tweag/cardano-peras/issues/120
-  case traverse (validatePerasCert perasCfg) certs of
+  case traverse (validatePerasCert perasParams) certs of
     Left validationErr -> throw (PerasCertValidationError validationErr)
     Right validatedCerts -> return validatedCerts
+
+-- | Add a list of 'PerasCert's into an object pool.
+--
+-- NOTE: we first validate the certificates, throwing an exception if any of
+-- them are invalid. We then wrap them with their arrival time, and finally add
+-- them to the pool using the provided adder function.
+--
+-- The order of the first two operations (i.e., validation and timestamping) are
+-- rather arbitrary, and the abstract Peras protocol just assumes it can happen
+-- "within" a slot.
+addPerasCerts ::
+  (StandardHash blk, MonadThrow m) =>
+  SystemTime m ->
+  (WithArrivalTime (ValidatedPerasCert blk) -> m a) ->
+  [PerasCert blk] ->
+  m ()
+addPerasCerts systemTime adder = do
+  validatePerasCerts
+    >=> mapM (addArrivalTime systemTime)
```

Comment:
This symbol makes me have to re-read the function a couple of times. I think it is easily avoidable, no?
```
addPerasCerts systemTime adder certs = do
  validatePerasCerts certs >>= mapM (addArrivalTime systemTime) >>= mapM_ adder
```